File: Python/BinarySearch/BinarySearch.py
def binary_search(arr, value, *, start = 0, end = None):
	if not end:
		end = len(arr)

	length = (end - start)

	if length <= 0:
		return False

	pos = start + (length // 2)

	if arr[pos] == value:
		return pos
	elif arr[pos] > value:
		return binary_search(arr, value, start = start, end = pos)
	elif arr[pos] < value:
		return binary_search(arr, value, start = pos+1, end = end)

File: Ruby/BinarySearch/BinarySearch.rb
def binary_search(arr, value, from: 0, to: nil)
  to = arr.length if to.nil?

  len = (to - from)

  if len <= 0 then
    return false
  end

  pos = from + (len / 2)

  if arr[pos] == value then
    return pos
  elsif arr[pos] > value then
    return binary_search(arr, value, from: from, to: pos)
  elsif arr[pos] < value then
    return binary_search(arr, value, from: pos+1, to: to)
  end
end
